### ISSUE TITLE
bug: Megasync service core dump.

### ISSUE THREAD

**Issue by knokelmaat** [nix-community/home-manager#5889]
### Are you following the right branch?

- [X] My Nixpkgs and Home Manager versions are in sync

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Issue description

Megasync is enabled in Home Manager using `services.megasync.enable`, but does not launch correctly on login.

Relevant logs:
```
Sep 24 09:39:40 nixos-B450M systemd[2091]: Started megasync.
Sep 24 09:39:40 nixos-B450M megasync[2177]: Avoiding wayland
Sep 24 09:39:40 nixos-B450M megasync[2177]: Error for "xrdb -query" command: "xrdb: Can't open display ''\n"
Sep 24 09:39:40 nixos-B450M megasync[2177]: qt.qpa.xcb: could not connect to display
Sep 24 09:39:40 nixos-B450M megasync[2177]: qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
Sep 24 09:39:40 nixos-B450M megasync[2177]: This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.
Sep 24 09:39:40 nixos-B450M megasync[2177]: Available platform plugins are: wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx, eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, xcb.
Sep 24 09:39:40 nixos-B450M systemd-coredump[2221]: Process 2177 (.megasync-wrapp) of user 1000 terminated abnormally with signal 6/ABRT, processing...
Sep 24 09:39:40 nixos-B450M systemd-coredump[2229]: Process 2177 (.megasync-wrapp) of user 1000 dumped core.
                                                    
                                                    Module libxcb-util.so.1 without build-id.
                                                    Module libexpat.so.1 without build-id.
                                                    Module libxkbcommon.so.0 without build-id.
                                                    Module libxkbcommon-x11.so.0 without build-id.
                                                    Module libICE.so.6 without build-id.
                                                    Module libSM.so.6 without build-id.
                                                    Module libxcb-xinput.so.0 without build-id.
                                                    Module libxcb-xkb.so.1 without build-id.
                                                    Module libxcb-xinerama.so.0 without build-id.
                                                    Module libxcb-xfixes.so.0 without build-id.
                                                    Module libxcb-sync.so.1 without build-id.
                                                    Module libxcb-shape.so.0 without build-id.
                                                    Module libxcb-render.so.0 without build-id.
                                                    Module libxcb-render-util.so.0 without build-id.
                                                    Module libxcb-randr.so.0 without build-id.
                                                    Module libxcb-keysyms.so.1 without build-id.
                                                    Module libxcb-shm.so.0 without build-id.
                                                    Module libxcb-image.so.0 without build-id.
                                                    Module libxcb-icccm.so.4 without build-id.
                                                    Module libX11-xcb.so.1 without build-id.
                                                    Module libfontconfig.so.1 without build-id.
                                                    Module libQt5XcbQpa.so.5 without build-id.
                                                    Module libqxcb.so without build-id.
                                                    Module libffi.so.8 without build-id.
                                                    Module libatomic.so.1 without build-id.
                                                    Module libpcre2-8.so.0 without build-id.
                                                    Module libgraphite2.so.3 without build-id.
                                                    Module libfreetype.so.6 without build-id.
                                                    Module libbrotlicommon.so.1 without build-id.
                                                    Module libkeyutils.so.1 without build-id.
                                                    Module libkrb5support.so.0 without build-id.
                                                    Module libcom_err.so.3 without build-id.
                                                    Module libk5crypto.so.3 without build-id.
                                                    Module libkrb5.so.3 without build-id.
                                                    Module libgmp.so.10 without build-id.
                                                    Module libhogweed.so.6 without build-id.
                                                    Module libnettle.so.8 without build-id.
                                                    Module libtasn1.so.6 without build-id.
                                                    Module libunistring.so.5 without build-id.
                                                    Module libp11-kit.so.0 without build-id.
                                                    Module libnuma.so.1 without build-id.
                                                    Module libogg.so.0 without build-id.
                                                    Module libvmaf.so.3 without build-id.
                                                    Module libsharpyuv.so.0 without build-id.
                                                    Module libGLdispatch.so.0 without build-id.
                                                    Module libXext.so.6 without build-id.
                                                    Module libGLX.so.0 without build-id.
                                                    Module libpcre2-16.so.0 without build-id.
                                                    Module libicudata.so.74 without build-id.
                                                    Module libicuuc.so.74 without build-id.
                                                    Module libicui18n.so.74 without build-id.
                                                    Module libharfbuzz.so.0 without build-id.
                                                    Module libpng16.so.16 without build-id.
                                                    Module libXdmcp.so.6 without build-id.
                                                    Module libXau.so.6 without build-id.
                                                    Module libcap.so.2 without build-id.
                                                    Module libbrotlidec.so.1 without build-id.
                                                    Module libzstd.so.1 without build-id.
                                                    Module libgssapi_krb5.so.2 without build-id.
                                                    Module libpsl.so.5 without build-id.
                                                    Module libssh2.so.1 without build-id.
                                                    Module libidn2.so.0 without build-id.
                                                    Module libnghttp2.so.14 without build-id.
                                                    Module libsoxr.so.0 without build-id.
                                                    Module libdrm.so.2 without build-id.
                                                    Module libX11.so.6 without build-id.
                                                    Module libvdpau.so.1 without build-id.
                                                    Module libva-drm.so.2 without build-id.
                                                    Module libssh.so.4 without build-id.
                                                    Module libsrt.so.1.5 without build-id.
                                                    Module libgnutls.so.30 without build-id.
                                                    Module libbz2.so.1 without build-id.
                                                    Module libva.so.2 without build-id.
                                                    Module libxvidcore.so.4 without build-id.
                                                    Module libx265.so.209 without build-id.
                                                    Module libx264.so.164 without build-id.
                                                    Module libwebp.so.7 without build-id.
                                                    Module libvorbisenc.so.2 without build-id.
                                                    Module libvorbis.so.0 without build-id.
                                                    Module libtheoradec.so.1 without build-id.
                                                    Module libtheoraenc.so.1 without build-id.
                                                    Module libSvtAv1Enc.so.2 without build-id.
                                                    Module libspeex.so.1 without build-id.
                                                    Module libopus.so.0 without build-id.
                                                    Module libopenjp2.so.7 without build-id.
                                                    Module libmp3lame.so.0 without build-id.
                                                    Module libaom.so.3 without build-id.
                                                    Module libdav1d.so.7 without build-id.
                                                    Module libcelt0.so.2 without build-id.
                                                    Module liblzma.so.5 without build-id.
                                                    Module libwebpmux.so.3 without build-id.
                                                    Module libvpx.so.9 without build-id.
                                                    Module libvulkan.so without build-id.
                                                    Module liblcms2.so.2 without build-id.
                                                    Module libjpeg.so.62 without build-id.
                                                    Module libgcc_s.so.1 without build-id.
                                                    Module libgomp.so.1 without build-id.
                                                    Module libstdc++.so.6 without build-id.
                                                    Module libGL.so.1 without build-id.
                                                    Module libQt5Core.so.5 without build-id.
                                                    Module libQt5DBus.so.5 without build-id.
                                                    Module libQt5Network.so.5 without build-id.
                                                    Module libQt5Gui.so.5 without build-id.
                                                    Module libQt5X11Extras.so.5 without build-id.
                                                    Module libQt5Widgets.so.5 without build-id.
                                                    Module libQt5Svg.so.5 without build-id.
                                                    Module libxcb.so.1 without build-id.
                                                    Module libcares.so.2 without build-id.
                                                    Module libcryptopp.so.8 without build-id.
                                                    Module libz.so.1 without build-id.
                                                    Module libswresample.so.5 without build-id.
                                                    Module libswscale.so.8 without build-id.
                                                    Module libavutil.so.59 without build-id.
                                                    Module libavformat.so.61 without build-id.
                                                    Module libavcodec.so.61 without build-id.
                                                    Module libraw.so.23 without build-id.
                                                    Module libzen.so.0 without build-id.
                                                    Module libmediainfo.so.0 without build-id.
                                                    Module .megasync-wrapped without build-id.
                                                    Stack trace of thread 2177:
                                                    #0  0x00007f1095a9b7dc __pthread_kill_implementation (libc.so.6 + 0x927dc)
                                                    #1  0x00007f1095a49516 raise (libc.so.6 + 0x40516)
                                                    #2  0x00007f1095a31935 abort (libc.so.6 + 0x28935)
                                                    #3  0x00007f109609a11d _ZNK14QMessageLogger5fatalEPKcz (libQt5Core.so.5 + 0x9a11d)
                                                    #4  0x00007f1096928290 _ZN22QGuiApplicationPrivate25createPlatformIntegrationEv (libQt5Gui.so.5 + 0x128290)
                                                    #5  0x00007f1096928828 _ZN22QGuiApplicationPrivate21createEventDispatcherEv (libQt5Gui.so.5 + 0x128828)
                                                    #6  0x00007f10962f0f35 _ZN23QCoreApplicationPrivate4initEv (libQt5Core.so.5 + 0x2f0f35)
                                                    #7  0x00007f109692b73c _ZN22QGuiApplicationPrivate4initEv (libQt5Gui.so.5 + 0x12b73c)
                                                    #8  0x00007f109692c774 _ZN15QGuiApplicationC2ERiPPci (libQt5Gui.so.5 + 0x12c774)
                                                    #9  0x00000000009f5fad n/a (.megasync-wrapped + 0x5f5fad)
                                                    #10 0x00000000009f653f n/a (.megasync-wrapped + 0x5f653f)
                                                    #11 0x00000000004ecded n/a (.megasync-wrapped + 0xecded)
                                                    #12 0x00007f1095a3314e __libc_start_call_main (libc.so.6 + 0x2a14e)
                                                    #13 0x00007f1095a33209 __libc_start_main@@GLIBC_2.34 (libc.so.6 + 0x2a209)
                                                    #14 0x00000000004ef2a5 n/a (.megasync-wrapped + 0xef2a5)
                                                    
                                                    Stack trace of thread 2220:
                                                    #0  0x00007f1095b0ad1f __poll (libc.so.6 + 0x101d1f)
                                                    #1  0x00007f1091d1121f g_main_context_iterate_unlocked.isra.0 (libglib-2.0.so.0 + 0x6121f)
                                                    #2  0x00007f1091d118ec g_main_context_iteration (libglib-2.0.so.0 + 0x618ec)
                                                    #3  0x00007f1096345c86 _ZN20QEventDispatcherGlib13processEventsE6QFlagsIN10QEventLoop17ProcessEventsFlagEE (libQt5Core.so.5 + 0x345c86)
                                                    #4  0x00007f10962e86eb _ZN10QEventLoop4execE6QFlagsINS_17ProcessEventsFlagEE (libQt5Core.so.5 + 0x2e86eb)
                                                    #5  0x00007f10960e3663 _ZN7QThread4execEv (libQt5Core.so.5 + 0xe3663)
                                                    #6  0x00007f1097d847c7 _ZN22QDBusConnectionManager3runEv (libQt5DBus.so.5 + 0x197c7)
                                                    #7  0x00007f10960e4aec _ZN14QThreadPrivate5startEPv (libQt5Core.so.5 + 0xe4aec)
                                                    #8  0x00007f1095a99a42 start_thread (libc.so.6 + 0x90a42)
                                                    #9  0x00007f1095b1905c __clone3 (libc.so.6 + 0x11005c)
                                                    ELF object binary architecture: AMD x86-64
Sep 24 09:39:40 nixos-B450M systemd[2091]: megasync.service: Main process exited, code=dumped, status=6/ABRT
Sep 24 09:39:40 nixos-B450M systemd[2091]: megasync.service: Failed with result 'core-dump'.

```

### System information

```markdown
- system: `"x86_64-linux"`
 - host os: `Linux 6.6.52, NixOS, 24.11 (Vicuna), 24.11.20240921.9357f4f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.7`
 - channels(root): `"nixos-24.05"`
 - nixpkgs: `/nix/store/y6205wq8hxvpqvl8l9d1n9xah01kg0lq-source`
```

**Comment by arakan94** [comment]
Also have this issue when using GNOME on Wayland. On X11, it seems to cause whole login to fail (see #5910).

Using proprietary Nvidia drivers (1050 Ti), on fresh install of 24.05..

```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.52, NixOS, 24.05 (Uakari), 24.05.20240927.f651414`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.8`
 - channels(root): `"home-manager-24.05.tar.gz, nixos-24.05"`
 - nixpkgs: `/nix/store/2vmkhm30kdnics1bnj9zgblh9yr4cpyw-source`
```

Log:
```
zář 29 13:58:08 nixos-david .gnome-shell-wr[2979]: Running GNOME Shell (using mutter 46.2) as a Wayland display server
zář 29 13:58:08 nixos-david rtkit-daemon[1357]: Successfully made thread 3028 of process 2979 owned by '1000' RT at priority 20.
zář 29 13:58:08 nixos-david .gnome-shell-wr[2979]: Made thread 'KMS thread' realtime scheduled
zář 29 13:58:08 nixos-david systemd-coredump[2980]: [🡕] Process 2936 (.megasync-wrapp) of user 1000 dumped core.
                                                      
                                                      Module libxcb-util.so.1 without build-id.
                                                      Module libexpat.so.1 without build-id.
                                                      Module libxkbcommon.so.0 without build-id.
                                                      Module libxkbcommon-x11.so.0 without build-id.
                                                      Module libICE.so.6 without build-id.
                                                      Module libSM.so.6 without build-id.
                                                      Module libxcb-xinput.so.0 without build-id.
                                                      Module libxcb-xkb.so.1 without build-id.
                                                      Module libxcb-xinerama.so.0 without build-id.
                                                      Module libxcb-xfixes.so.0 without build-id.
                                                      Module libxcb-sync.so.1 without build-id.
                                                      Module libxcb-shape.so.0 without build-id.
                                                      Module libxcb-render.so.0 without build-id.
                                                      Module libxcb-render-util.so.0 without build-id.
                                                      Module libxcb-randr.so.0 without build-id.
                                                      Module libxcb-keysyms.so.1 without build-id.
                                                      Module libxcb-shm.so.0 without build-id.
                                                      Module libxcb-image.so.0 without build-id.
                                                      Module libxcb-icccm.so.4 without build-id.
                                                      Module libX11-xcb.so.1 without build-id.
                                                      Module libfontconfig.so.1 without build-id.
                                                      Module libQt5XcbQpa.so.5 without build-id.
                                                      Module libqxcb.so without build-id.
                                                      Module libffi.so.8 without build-id.
                                                      Module libatomic.so.1 without build-id.
                                                      Module libpcre2-8.so.0 without build-id.
                                                      Module libsystemd.so.0 without build-id.
                                                      Module libgraphite2.so.3 without build-id.
                                                      Module libfreetype.so.6 without build-id.
                                                      Module libbrotlicommon.so.1 without build-id.
                                                      Module libkeyutils.so.1 without build-id.
                                                      Module libkrb5support.so.0 without build-id.
                                                      Module libcom_err.so.3 without build-id.
                                                      Module libk5crypto.so.3 without build-id.
                                                      Module libkrb5.so.3 without build-id.
                                                      Module libgmp.so.10 without build-id.
                                                      Module libhogweed.so.6 without build-id.
                                                      Module libnettle.so.8 without build-id.
                                                      Module libtasn1.so.6 without build-id.
                                                      Module libunistring.so.5 without build-id.
                                                      Module libp11-kit.so.0 without build-id.
                                                      Module libnuma.so.1 without build-id.
                                                      Module libogg.so.0 without build-id.
                                                      Module libGLdispatch.so.0 without build-id.
                                                      Module libXext.so.6 without build-id.
                                                      Module libGLX.so.0 without build-id.
                                                      Module libpcre2-16.so.0 without build-id.
                                                      Module libicudata.so.73 without build-id.
                                                      Module libicuuc.so.73 without build-id.
                                                      Module libicui18n.so.73 without build-id.
                                                      Module libharfbuzz.so.0 without build-id.
                                                      Module libpng16.so.16 without build-id.
                                                      Module libXdmcp.so.6 without build-id.
                                                      Module libXau.so.6 without build-id.
                                                      Module libcap.so.2 without build-id.
                                                      Module libbrotlidec.so.1 without build-id.
                                                      Module libzstd.so.1 without build-id.
                                                      Module libgssapi_krb5.so.2 without build-id.
                                                      Module libpsl.so.5 without build-id.
                                                      Module libssh2.so.1 without build-id.
                                                      Module libidn2.so.0 without build-id.
                                                      Module libnghttp2.so.14 without build-id.
                                                      Module libsoxr.so.0 without build-id.
                                                      Module libvulkan.so.1 without build-id.
                                                      Module libdrm.so.2 without build-id.
                                                      Module libX11.so.6 without build-id.
                                                      Module libvdpau.so.1 without build-id.
                                                      Module libva-drm.so.2 without build-id.
                                                      Module libssh.so.4 without build-id.
                                                      Module libsrt.so.1.5 without build-id.
                                                      Module libgnutls.so.30 without build-id.
                                                      Module libbz2.so.1 without build-id.
                                                      Module libva.so.2 without build-id.
                                                      Module libxvidcore.so.4 without build-id.
                                                      Module libx265.so.199 without build-id.
                                                      Module libx264.so.164 without build-id.
                                                      Module libvorbisenc.so.2 without build-id.
                                                      Module libvorbis.so.0 without build-id.
                                                      Module libtheoradec.so.1 without build-id.
                                                      Module libtheoraenc.so.1 without build-id.
                                                      Module libSvtAv1Enc.so.2 without build-id.
                                                      Module libspeex.so.1 without build-id.
                                                      Module libopus.so.0 without build-id.
                                                      Module libmp3lame.so.0 without build-id.
                                                      Module libdav1d.so.7 without build-id.
                                                      Module liblzma.so.5 without build-id.
                                                      Module libvpx.so.9 without build-id.
                                                      Module liblcms2.so.2 without build-id.
                                                      Module libjpeg.so.62 without build-id.
                                                      Module libgcc_s.so.1 without build-id.
                                                      Module libgomp.so.1 without build-id.
                                                      Module libstdc++.so.6 without build-id.
                                                      Module libGL.so.1 without build-id.
                                                      Module libQt5Core.so.5 without build-id.
                                                      Module libQt5DBus.so.5 without build-id.
                                                      Module libQt5Network.so.5 without build-id.
                                                      Module libQt5Gui.so.5 without build-id.
                                                      Module libQt5X11Extras.so.5 without build-id.
                                                      Module libQt5Widgets.so.5 without build-id.
                                                      Module libQt5Svg.so.5 without build-id.
                                                      Module libxcb.so.1 without build-id.
                                                      Module libudev.so.1 without build-id.
                                                      Module libcares.so.2 without build-id.
                                                      Module libcryptopp.so.8 without build-id.
                                                      Module libz.so.1 without build-id.
                                                      Module libswresample.so.3 without build-id.
                                                      Module libswscale.so.5 without build-id.
                                                      Module libavutil.so.56 without build-id.
                                                      Module libavformat.so.58 without build-id.
                                                      Module libavcodec.so.58 without build-id.
                                                      Module libraw.so.23 without build-id.
                                                      Module libzen.so.0 without build-id.
                                                      Module libmediainfo.so.0 without build-id.
                                                      Module .megasync-wrapped without build-id.
                                                      Stack trace of thread 2936:
                                                      #0  0x00007fa6f2ca2efc __pthread_kill_implementation (libc.so.6 + 0x8fefc)
                                                      #1  0x00007fa6f2c52e86 raise (libc.so.6 + 0x3fe86)
                                                      #2  0x00007fa6f2c3b935 abort (libc.so.6 + 0x28935)
                                                      #3  0x00007fa6f329a0a7 _ZNK14QMessageLogger5fatalEPKcz (libQt5Core.so.5 + 0x9a0a7)
                                                      #4  0x00007fa6f3b26c70 _ZN22QGuiApplicationPrivate25createPlatformIntegrationEv (libQt5Gui.so.5 + 0x126c70)
                                                      #5  0x00007fa6f3b27200 _ZN22QGuiApplicationPrivate21createEventDispatcherEv (libQt5Gui.so.5 + 0x127200)
                                                      #6  0x00007fa6f34e2ba5 _ZN23QCoreApplicationPrivate4initEv (libQt5Core.so.5 + 0x2e2ba5)
                                                      #7  0x00007fa6f3b2a07c _ZN22QGuiApplicationPrivate4initEv (libQt5Gui.so.5 + 0x12a07c)
                                                      #8  0x00007fa6f3b2b0a4 _ZN15QGuiApplicationC2ERiPPci (libQt5Gui.so.5 + 0x12b0a4)
                                                      #9  0x00000000009e00ad n/a (.megasync-wrapped + 0x5e00ad)
                                                      #10 0x00000000009e063f n/a (.megasync-wrapped + 0x5e063f)
                                                      #11 0x00000000004ecd3d n/a (.megasync-wrapped + 0xecd3d)
                                                      #12 0x00007fa6f2c3d10e __libc_start_call_main (libc.so.6 + 0x2a10e)
                                                      #13 0x00007fa6f2c3d1c9 __libc_start_main@@GLIBC_2.34 (libc.so.6 + 0x2a1c9)
                                                      #14 0x00000000004ef1f5 n/a (.megasync-wrapped + 0xef1f5)
                                                      
                                                      Stack trace of thread 2963:
                                                      #0  0x00007fa6f2d0f62f __poll (libc.so.6 + 0xfc62f)
                                                      #1  0x00007fa6ef880f2f g_main_context_iterate_unlocked.isra.0 (libglib-2.0.so.0 + 0x5ef2f)
                                                      #2  0x00007fa6ef8815cc g_main_context_iteration (libglib-2.0.so.0 + 0x5f5cc)
                                                      #3  0x00007fa6f3535256 _ZN20QEventDispatcherGlib13processEventsE6QFlagsIN10QEventLoop17ProcessEventsFlagEE (libQt5Core.so.5 + 0x335256)
                                                      #4  0x00007fa6f34da6eb _ZN10QEventLoop4execE6QFlagsINS_17ProcessEventsFlagEE (libQt5Core.so.5 + 0x2da6eb)
                                                      #5  0x00007fa6f32e24d3 _ZN7QThread4execEv (libQt5Core.so.5 + 0xe24d3)
                                                      #6  0x00007fa6f4f87767 _ZN22QDBusConnectionManager3runEv (libQt5DBus.so.5 + 0x19767)
                                                      #7  0x00007fa6f32e38ec _ZN14QThreadPrivate5startEPv (libQt5Core.so.5 + 0xe38ec)
                                                      #8  0x00007fa6f2ca1272 start_thread (libc.so.6 + 0x8e272)
                                                      #9  0x00007fa6f2d1cdec __clone3 (libc.so.6 + 0x109dec)
                                                      ELF object binary architecture: AMD x86-64
```

**Comment by GaetanLepage** [comment]
Can you test if https://github.com/nix-community/home-manager/pull/5902/ is fixing this issue ?
I am not using this module anymore (neither do I use gnome).
Is this a HM-related problem or a megasync packaging issue ?

**Comment by arakan94** [comment]
> Can you test if #5902 is fixing this issue ? I am not using this module anymore (neither do I use gnome). Is this a HM-related problem or a megasync packaging issue ?

Just tested it and yes, it fixes the issue completely.

I'd say it was HM issue, since installing the package via configuration and adding it to startup applications in Tweaks worked fine..

**Comment by GaetanLepage** [comment]
Fixed by #5902 (you may now close this issue).

**Comment by Ptitet** [comment]
I'am having the exact same issue with same logs as @knokelmaat. I'm using Niri on Wayland.

```
 - system: `"x86_64-linux"`
 - host os: `Linux 6.17.8, NixOS, 25.11 (Xantusia), 25.11.20251117.89c2b23`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2`
 - nixpkgs: `/nix/store/f640ps0hcp7w5jzg18djf8gdhl6r2rnl-source`
```
